apps/api/src/hyp3_api/validation.py

```diff
@@ -70,31 +70,38 @@ def check_granules_exist(granules, granule_metadata):
         raise GranuleValidationError(f'Some requested scenes could not be found: {", ".join(not_found_granules)}')
 
 
-def check_dem_coverage(granule_metadata):
+def check_dem_coverage(_, granule_metadata):
     bad_granules = [g['name'] for g in granule_metadata if not has_sufficient_coverage(g['polygon'])]
     if bad_granules:
         raise GranuleValidationError(f'Some requested scenes do not have DEM coverage: {", ".join(bad_granules)}')
 
 
-def check_same_burst_ids(granule_metadata):
-    ref_burst_id, sec_burst_id = [granule['name'].split('_')[1] for granule in granule_metadata]
-    if ref_burst_id != sec_burst_id:
+def check_same_burst_ids(job, _):
+    ref_ids = [ref.split('_')[1] for ref in job['job_parameters'].get('reference', [])]
+    sec_ids = [sec.split('_')[1] for sec in job['job_parameters'].get('secondary', [])]
+    if ref_ids != sec_ids:
         raise GranuleValidationError(
-            f'The requested scenes do not have the same burst ID: {ref_burst_id} and {sec_burst_id}'
+            'The requested scenes have burst IDs with no matching pairs.'
+        )
+    if len(set(ref_ids)) != len(ref_ids):
+        raise GranuleValidationError(
+            'The requested scenes have more than 1 pair with the same burst ID.'
         )
 
 
-def check_valid_polarizations(granule_metadata):
-    ref_polarization, sec_polarization = [granule['name'].split('_')[4] for granule in granule_metadata]
-    if ref_polarization != sec_polarization:
+def check_valid_polarizations(_, granule_metadata):
+    polarizations = set(granule['name'].split('_')[4] for granule in granule_metadata)
```

Comment:
Let's use the granules from `job` unless we specifically need the extra information from `granule_metadata`. This makes it more obvious which validators actually rely on CMR metadata, which may help use with future refactoring efforts.
```suggestion
def check_valid_polarizations(job, _):
    polarizations = set(granule.split('_')[4] for granule in get_granules([job]))
```